File: python-solutions/two-sum.py
def twoSum(nums, target):
    numMap = {}
    i = 0;

    while i < len(nums):
        complement = target - nums[i]
        if complement in numMap:
            return [i,numMap[complement]]
        numMap[nums[i]] = i
        i += 1
    return []


# TEST CASES
nums_one = [1,2,3,4,4,5];
nums_two = [3,4,5,8,2];
nums_three = [9,4,5,7,9,4];
print(twoSum(nums_one, 7))
print(twoSum(nums_two, 11))
print(twoSum(nums_three, 23))

File: TypeScript/twoSum.ts
const twoSum = (nums: number[], target: number ): number[] => {
    const map: Record<number, number> = {};

    for (let i: number = 0; i < nums.length; i += 1) {
        let complement: number = target - nums[i];

        if (complement in map) {
            return [map[complement], i];
        }
        map[nums[i]] = i;
    }
    return [];
};


const testOne: number[] = [1,2,3,4,4,5];
const testTwo: number[] = [3,4,5,8,2];
const testThree: number[] = [9,4,5,7,9,4];

console.log(`Output: ${twoSum(testOne, 7)}`);
console.log(`Output: ${twoSum(testTwo, 34)}`);
console.log(`Output: ${twoSum(testThree, 13)}`);